src/main/java/io/kestra/plugin/nats/Request.java

```diff
@@ -0,0 +1,200 @@
+        """
+    )
+    @NotNull
+    @PluginProperty(dynamic = true)
+    private Object from;
+
+    @Schema(
+        title = "Timeout in milliseconds to wait for a response.",
+        description = "Defaults to 5000 ms."
+    )
+    @Builder.Default
+    private Integer requestTimeout = 5000;
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        // 1) Connect to NATS
+        Connection connection = this.connect(runContext);
+
+        // 2) Interpolate the subject (if it has placeholders like {{ ... }})
+        String renderedSubject = runContext.render(this.subject);
+
+        // 3) Get the raw message from 'from' and then interpolate the entire map, including headers
+        Map<String, Object> rawMessage = retrieveSingleMessage(runContext);
+        Map<String, Object> renderedMessage = runContext.render(rawMessage);
+
+        // 4) Build the NATS message with the fully rendered subject + map
+        Message natsMessage = buildRequestMessage(renderedSubject, renderedMessage);
+
+        // 5) Execute request-reply with the configured timeout
+        Duration timeoutDuration = Duration.ofMillis(this.requestTimeout);
+        Message reply = connection.request(natsMessage, timeoutDuration);
+
+        // 6) Convert the reply (if any) to a UTF-8 string
+        String response = (reply == null) ? null : new String(reply.getData(), StandardCharsets.UTF_8);
+
+        connection.close();
+
+        return Output.builder()
+            .response(response)
+            .build();
+    }
+
+    @SuppressWarnings("unchecked")
+    private Map<String, Object> retrieveSingleMessage(RunContext runContext) throws Exception {
+        switch (this.from) {
+            // A single Kestra storage URI
+            case String fromStr -> {
+                URI fromUri = new URI(runContext.render(fromStr));
+                if (!"kestra".equalsIgnoreCase(fromUri.getScheme())) {
+                    throw new IllegalArgumentException("Invalid 'from': must be a kestra:// URI.");
+                }
+
+                // Parse the first record from the Ion/JSON file
+                try (BufferedReader reader = new BufferedReader(new InputStreamReader(runContext.storage().getFile(fromUri)))) {
+                    Flux<Object> flux = FileSerde.readAll(reader);
+                    Object first = flux.blockFirst(); // read the first object
+                    if (first == null) {
+                        throw new IllegalArgumentException("'from' file is empty; expected exactly one map.");
+                    }
+                    if (!(first instanceof Map)) {
+                        throw new IllegalArgumentException("'from' file must contain a map for the request message.");
+                    }
+                    return (Map<String, Object>) first;
+                }
+            }
+
+            // A list => only the first item is used
```

Comment:
Again, why supporting a list and only reading the first object?
The other objects would be lost.

Reply:
That is an oversight. I'll fix this.